include/tvm/tir/data_layout.h

```diff
@@ -157,6 +158,16 @@ class Layout : public ObjectRef {
     return undef;
   }
 
+  /*!
+   * \brief The layout to represent to represent the initial layout of a tensor
+   * with dynamic shape (unknown ndim specifically)
+   * \return a (global) dynamic initial layout.
+   */
+  static const Layout& InitialDynamic() {
+    static Layout initial_dynamic("__init_dyn__");
```

Comment:
What is the motivation of introducing a variant here? GIven it is really specific, it might be useful to handle this in convert layout, or leverage undef instead of introducing this layout variat.

Reply:
Undef is used for arguments that are not tensors. It is technically doable to use it for ndim-unknown tensors as well. But I feel it's a bit strange.

Reply:
Main thing is to avoid data layout format change

Reply:
what do you mean by format change?